docs/devguide/metricset-details.asciidoc

```diff
@@ -204,13 +205,96 @@ waitFor ${REDIS_HOST} ${REDIS_PORT} Redis
 The environment expects your service to be available as soon as it receives a response from
 the given address and port.
 
+[float]
+===== Adding the standard metricset integration tests
+
+There are normally two integration tests that are part of every metricset: `TestFetch` and `TestData`.
+Both tests will start up a new instance of your metricset and fetch an event. In order to start a metricset, you need to create a configuration object:
+
+[source,go]
+----
+func getConfig() map[string]interface{} {
+	return map[string]interface{}{
+		"module":           "{module}",
+		"metricsets":       []string{"{metricset}"},
+                "hosts":      []string{GetEnvHost() + ":" + GetEnvPort()}, <1>
+	}
+}
+
+func GetEnvHost() string { <2>
+	host := os.Getenv("{module}_HOST")
+	if len(host) == 0 {
+		host = "127.0.0.1"
+	}
+	return host
+}
+
+func GetEnvPort() string { <2>
+	port := os.Getenv("{module}_PORT")
+
+	if len(port) == 0 {
+		port = "1234"
+	}
+	return port
+}
+
+----
+<1> Add any additional config options your metricset needs here.
+<2> The endpoint used by the metricset needs to be configurable for manual and automated testing.
+
+The `TestFetch` integration test will return a single event from your metricset, which you can use to test the validity of the data.
+`TestData` will (re)generate the `_meta/data.json` file that documents the data reported by the metricset.
+
+[source,go]
+----
+import (
+	"os"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+
+	"github.com/elastic/beats/libbeat/tests/compose"
+	mbtest "github.com/elastic/beats/metricbeat/mb/testing"
+)
+
+func TestFetch(t *testing.T) {
+	compose.EnsureUp(t, "{module}") <1>
+
+	f := mbtest.NewReportingMetricSetV2Error(t, getConfig())
```

Comment:
I feel like it might be worth it just updating the docs once that PR is merged.

Considering that'll give us some 'generic' methods for a lot of this, I wonder if it's worth also adding a `_test.go` file to the generator.